cpp/modules/deck.gl/core/src/lib/deck.cpp

```diff
@@ -62,7 +62,7 @@ auto Deck::Props::getProperties() const -> const Properties* {
 
 Deck::Deck(std::shared_ptr<Deck::Props> props)
     : Component(props), width{props->width}, height{props->height}, _needsRedraw{"Initial render"} {
-  this->animationLoop = this->_createAnimationLoop(props);
+  this->animationLoop = lumagl::AnimationLoopFactory::createAnimationLoop(props->drawingOptions);
```

Comment:
Nit: Not a big fan of the factory pattern, or any construction that overrides the languages built in syntax for creating and initializing objects. Let's make an issue for API audit and add a bullet for this.

Reply:
Will do. The idea was that I wanted to have logic for creating an animation loop, based on what backends are enabled in one place, and outside the `Deck` class. Once we figure out what this should ideally look like, it should be easy enough to revisit